backend/src/EventSubscriber/FishbowlStartIntroSubscriber.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Stooa codebase.
+ *
+ * (c) 2020 - present Runroom SL
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\EventSubscriber;
+
+use App\Entity\Fishbowl;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Workflow\Event\GuardEvent;
+use Webmozart\Assert\Assert;
+
+class FishbowlStartIntroSubscriber implements EventSubscriberInterface
```

Comment:
Do we need 2 classes? doesn't the GuardEvent include some information about the current transition so we can check if the transition is one of the two defined?

Reply:
The logic to block the workflow is different depending the transition. I don't feel like we are repeating the code

Reply:
The only difference between those 2 classes is a ! on the if. To me it looks like it could be handled with only one subscriber that decides if the `if` is negated or not.

Reply:
recheck!